src/ui/pages/zcl_abapgit_gui_page_merge.clas.abap

```diff
@@ -1,68 +1,60 @@
 CLASS zcl_abapgit_gui_page_merge DEFINITION
   PUBLIC
-  INHERITING FROM zcl_abapgit_gui_page
+  INHERITING FROM zcl_abapgit_gui_component
   FINAL
-  CREATE PUBLIC .
+  CREATE PRIVATE.
 
   PUBLIC SECTION.
 
+    INTERFACES:
+      zif_abapgit_gui_event_handler,
+      zif_abapgit_gui_menu_provider,
+      zif_abapgit_gui_renderable.
+
+    CLASS-METHODS create
+      IMPORTING
+        io_repo        TYPE REF TO zcl_abapgit_repo_online
+        iv_source      TYPE string
+        iv_target      TYPE string
+      RETURNING
+        VALUE(ri_page) TYPE REF TO zif_abapgit_gui_renderable
+      RAISING
+        zcx_abapgit_exception.
+
     METHODS constructor
       IMPORTING
         io_repo   TYPE REF TO zcl_abapgit_repo_online
         iv_source TYPE string
         iv_target TYPE string
       RAISING
-        zcx_abapgit_exception .
+        zcx_abapgit_exception.
 
-    METHODS zif_abapgit_gui_event_handler~on_event
-        REDEFINITION.
   PROTECTED SECTION.
-    METHODS render_content REDEFINITION.
-
   PRIVATE SECTION.
 
-    DATA mo_repo TYPE REF TO zcl_abapgit_repo_online .
-    DATA mi_merge TYPE REF TO zif_abapgit_merge .
+    DATA mo_repo TYPE REF TO zcl_abapgit_repo_online.
+    DATA mi_merge TYPE REF TO zif_abapgit_merge.
+
     CONSTANTS:
       BEGIN OF c_actions,
-        merge         TYPE string VALUE 'merge' ##NO_TEXT,
-        res_conflicts TYPE string VALUE 'res_conflicts' ##NO_TEXT,
-      END OF c_actions .
+        merge         TYPE string VALUE 'merge',
+        res_conflicts TYPE string VALUE 'res_conflicts',
+      END OF c_actions.
```

Comment:
no need to actively decide to change these lines, SE80 will change it back again for some other developer

Reply:
until we all use vsc